cmd/stack/deploy.go

```diff
@@ -0,0 +1,169 @@
+			s, err := utils.LoadStack(args[0], stackPath)
+			if err != nil {
+				return err
+			}
+
+			if err := s.UpdateNamespace(namespace); err != nil {
+				return err
+			}
+			err = executeDeployStack(ctx, s, forceBuild, wait)
+			analytics.TrackDeployStack(err == nil)
+			if err == nil {
+				log.Success("Successfully deployed stack '%s'", s.Name)
+			}
+			return err
+		},
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return fmt.Errorf("deploy requires the stack NAME argument")
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVarP(&stackPath, "file", "f", utils.DefaultStackManifest, "path to the stack manifest file")
+	cmd.Flags().StringVarP(&namespace, "namespace", "n", "", "overwrites the stack namespace where the stack is deployed")
+	cmd.Flags().BoolVarP(&forceBuild, "build", "", false, "build images before starting containers")
+	cmd.Flags().BoolVarP(&wait, "wait", "", false, "wait until a minimum number of Pods of a Deployment, StatefulSet are in a ready state")
+	return cmd
+}
+
+func executeDeployStack(ctx context.Context, s *model.Stack, forceBuild, wait bool) error {
+	settings := cli.New()
+	if s.Namespace == "" {
+		s.Namespace = settings.Namespace()
+	}
+
+	if err := helm.Translate(s, forceBuild); err != nil {
+		return err
+	}
+
+	dynamicStackFilename, err := saveStackFile(s)
+	if err != nil {
+		return err
+	}
+	defer os.Remove(dynamicStackFilename)
+
+	valueOpts := &values.Options{}
+	valueOpts.ValueFiles = []string{dynamicStackFilename}
+	vals, err := valueOpts.MergeValues(nil)
+	if err != nil {
+		return fmt.Errorf("error initializing stack values: %s", err)
+	}
+
+	var re *repo.Entry
+	rf, err := repo.LoadFile(settings.RepositoryConfig)
+	if !isNotExist(err) {
+		for _, r := range rf.Repositories {
+			if r.Name != HelmRepoName {
+				continue
+			}
+			re = r
+			break
+		}
+	}
+	if re == nil {
+		if err := helm.RepoAdd(settings, HelmRepoName, HelmRepoURL, HelmChartName, HelmChartVersion); err != nil {
+			return err
+		}
+		log.Information("'%s' has been added to your helm repositories.", HelmRepoName)
+	} else {
+		if err := helm.RepoUpdate(re, settings, HelmChartName); err != nil {
+			return err
+		}
+	}
+
+	spinner := utils.NewSpinner(fmt.Sprintf("Deploying stack '%s'...", s.Name))
+	spinner.Start()
+	defer spinner.Stop()
+
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(settings.RESTClientGetter(), s.Namespace, helm.HelmDriver, func(format string, v ...interface{}) {
+		message := fmt.Sprintf(format, v...)
+		spinner.Update(fmt.Sprintf("%s...", message))
+	}); err != nil {
+		return fmt.Errorf("error initializing stack client: %s", err)
+	}
+
+	exists, err := helm.ExistRelease(action.NewList(actionConfig), s.Name)
+	if err != nil {
+		return fmt.Errorf("error listing stacks: %s", err)
+	}
+	if exists {
+		return helm.Upgrade(action.NewUpgrade(actionConfig), settings, s, HelmRepoName, HelmChartName, HelmChartVersion, vals, wait)
```

Comment:
We should keep all the helm stuff inside of the helm package so the abstraction is cleaner and easier to understand.